### DISCUSSION TITLE
Asking for help and recommendations

### DISCUSSION THREAD

**Discussion by alexisahedo** [johntruckenbrodt/pyroSAR#210] [Q&A] [answered]
Dear @johntruckenbrodt,

I‘ll like to do a series a questions that have been hard to answer as I can’t barely find any useful information, as always I really appreciate all your hard work and help. 

Giving some context, I’m working with an Open Data Cube infrastructure, I’m trying to add Sentinel-1 RTC to our archive and make a time-series national wide for the study of Natural Resources. I’m expecting to preprocess all the information to ARD, index and ingest it, so I’m working with the preparation’s scripts and yaml metadata to EO3 format (https://datacube-core.readthedocs.io/en/latest/installation/dataset-documents.html), sadly I haven’t been able to find any good example of Sentinel-1 product registration in the ODC, all information focus on Landsat and the metadata available for both mission have variations. 

Questions:
1.	Is there a way to obtain the path (I believe it to be the Relative Orbit) and frame of the scenes within the metadata, as in the ASF? “provider specific unique identified for the same location for Landsat '{:03d}{:03d}'.format(path, row)”.

2.	As a clarification of the processing final goal. In theory the processed scenes to RTC gamma0 (`snap.geocode`) are basically a Normalized Radar Backscatter product, right? Does that mean that the RTC normalize the incidence angle of different orbits, making it possible to do mosaics of different orbits?

3.	Regarding the processing time. Is there a way to optimize processing speed for big data sets?

4.	Within PyroSAR is there any tool to make RGB raster images, R-VV[db], G-VH[db] & B[VV/VH].

5.	Could you give me recommendations for handling the ODC? I’m relatively new in the topic.

An apology if I’m asking for too much help. 

Saludos.
Alexis A.

**Comment by johntruckenbrodt** [comment, marked as answer]
Dear @alexisahedo,
thanks for starting this discussion, I'll be glad to answer. Basically all your questions are addressed by the new Sentinel-1 Normalised Radar Backscatter product (S1-NRB) that I've been developing for the last 1.5 years. We are currently in the hot phase of wrapping up the project so excuse me for my brevity. You can have a look at the processor which builds on pyroSAR:
https://github.com/SAR-ARD/S1_NRB
This product comes with extensive metadata in XML and STAC JSON format, which should be quite easy to ingest into ODC.
Some short comments on your questions:

1. you can get the relative orbit number from a scene object with this: `id.meta['orbitNumber_rel']`; a unique frame is not assigned to the products.
2. principally yes. The RTC correction described [here](https://doi.org/10.1109/Tgrs.2011.2120616) is used. However, as much as terrain effects are reduced, merging images from different orbits should be done with caution. There can be larger differences, in particular in rough terrain. [This approach](https://doi.org/10.1109/tgrs.2021.3055562) can be used to further reduce these effects.
3. pyroSAR's processing speed depends on the software running in the background. Currently it is as fast as it can get.
4. you can have a look at function [S1_NRB.nrb.create_rgb_vrt](https://s1-nrb.readthedocs.io/en/latest/api.html#S1_NRB.nrb.create_rgb_vrt) on how to express RGB composites in [GDAL VRT](https://gdal.org/drivers/raster/vrt.html) files.
5. Not really. I looked a little into it some years ago and came up with this [ODC extension](https://pyrosar.readthedocs.io/en/latest/pyroSAR.html#module-pyroSAR.datacube_util) of pyroSAR. The new STAC metadata that comes with the new S1-NRB product currently is all I need in terms of data cube functionality. See the documentation section on [exploring data cubes](https://s1-nrb.readthedocs.io/en/latest/examples/nrb_cube.html).

Cheers
John

**Reply by alexisahedo** [reply]
Dear @johntruckenbrodt,

Thank you for your prompt response and assistance, the information you shared is of my biggest interest. Sentinel-1 Normalised Radar Backscatter product (S1-NRB) is a project I’ll follow closely, as I always say I’m not trying to reinvent the wheel, so this new implementation fits perfectly in my requirements. 

If I may do a short comment about the quick view I gave the repository, I’ll advice to give the users the possibility to add their own national grids, as ours uses Albers Equal Area of 150k meters tile and would be of great help to be able to work with it prior the integration of the data sets to the ODC. 

I’ll keep in contact if other questions come up, I’m sorry for abusing your kindness. 

Saludos
Alexis A.
